src/dashboard/src/components/filesystem_ajax/views.py

```diff
@@ -1103,7 +1104,14 @@ def download_by_uuid(request, uuid, preview_file=False):
             % {"uuid": uuid},
         }
         return helpers.json_response(response, status_code=404)
-    relative_path = f.currentlocation.replace("%transferDirectory%", "")
+    # check if the transfer is in the backlog and if it's a bag
+    transfer = models.Transfer.objects.get(uuid=f.transfer.uuid)
```

Comment:
Is this a good use case for Django's [`select_related(*fields)`](https://docs.djangoproject.com/en/1.8/ref/models/querysets/#select-related)? I've learned about it recently!

```python
    f = models.File.objects.select_related('transfer').get(uuid=uuid)

Reply:
Ah nice!